tests/src/test/scala/dogs/ARangeTest.scala

```diff
@@ -0,0 +1,87 @@
+/**
```

Comment:
why is this in its own directory? can we move it to the parent directory with the rest of the tests and name it RangeTest.scala or RangeSpec.scala?